rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__SUBSCRIPTION_OPTIONS_HPP_
+#define RCLCPP__SUBSCRIPTION_OPTIONS_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp/intra_process_setting.hpp"
+#include "rclcpp/visibility_control.hpp"
+
+namespace rclcpp
+{
+
+/// Structure containing optional configuration for Subscriptions.
+template<typename Alloc = std::allocator<void>>
```

Comment:
I notice this, when looking at the changes in other packages, can you use `Allocator` here please? I know we use `Alloc` in a few places, but I'd like to change those too at some point, using the full name avoids abbreviation and brings it inline with the style the standard library uses, e.g. `std::base_string`: `class Allocator = std::allocator<CharT>` (https://en.cppreference.com/w/cpp/string/basic_string)

Reply:
Ok - I have updated to use the non-abbreviated version in everything I added in this PR